src/state_transition/block/process_execution_payload.zig

```diff
@@ -95,10 +96,10 @@ pub fn processExecutionPayload(
         return error.InvalidExecutionPayload;
     }
 
-    const payload_header = switch (body) {
-        .regular => |b| try b.executionPayload().toPayloadHeader(allocator),
-        .blinded => |b| b.executionPayloadHeader(),
-    };
-
-    state.setLatestExecutionPayloadHeader(payload_header);
+    var payload_header = state.latestExecutionPayloadHeader();
+    payload_header.deinit(allocator);
+    switch (body) {
+        .regular => |b| try b.executionPayload().toPayloadHeader(allocator, &payload_header),
+        .blinded => |b| try b.executionPayloadHeader().clone(allocator, &payload_header),
```

Comment:
this pattern is strange to me, why deiniting `payload_header` then cloning into it? not sure how it works
it's better to always extract to new payload_header and call the new safe `setLatestExecutionPayloadHeader()` instead